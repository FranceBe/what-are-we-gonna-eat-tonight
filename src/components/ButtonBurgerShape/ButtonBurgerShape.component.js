import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { spaces, colors, fontSizes } from 'styles/variables'

export const BurgerShape = styled.button`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;
  height: ${spaces.large};
  width: ${spaces.large};
  border-radius: ${spaces.large} ${spaces.large} ${spaces.medium} ${spaces.medium};
  background:
  linear-gradient(0deg,
  ${colors.primary_default} 15%,
  ${colors.secondary_default} 27%,
  ${colors.tertiary_default} 47%,
  ${colors.secondary_light} 45%,
  ${colors.secondary_default} 60%,
  ${colors.primary_default} 79%);
  cursor: pointer;
  :focus {
    outline: none;
  }
`
export const ButtonText = styled.span`
  background-color: rgba(255, 255, 255, 0.7);
  padding: ${spaces.small};
  margin-bottom: ${spaces.small};
  font-size: ${fontSizes.px_14}
`

export class ButtonBurgerShape extends React.PureComponent {
  static propTypes = {
    children: PropTypes.any.isRequired,
    onClick: PropTypes.func,
  }

  onActionClick = e => {
    if (this.props.onClick) {
      this.props.onClick()
    } else {
      e.preventDefault()
    }
  }

  render() {
    const { ...rest } = this.props
    return (
      <BurgerShape onClick={this.onActionClick} {...rest}>
        <ButtonText>{this.props.children}</ButtonText>
      </BurgerShape>
    )
  }
}

export default ButtonBurgerShape
