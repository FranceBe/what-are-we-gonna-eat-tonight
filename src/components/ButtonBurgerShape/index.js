import ButtonBurgerShapeComponent,
{ ButtonBurgerShape, BurgerShape, ButtonText }
  from 'components/ButtonBurgerShape/ButtonBurgerShape.component'

export { ButtonBurgerShape, BurgerShape, ButtonText }

export default ButtonBurgerShapeComponent
