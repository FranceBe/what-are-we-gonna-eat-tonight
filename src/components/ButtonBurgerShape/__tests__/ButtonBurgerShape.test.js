import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import ButtonBurgerShape,
{ BurgerShape, ButtonText }
  from 'components/ButtonBurgerShape/ButtonBurgerShape.component'

describe('ButtonBurgerShape', () => {
  describe('styled components', () => {
    describe('BurgerShape', () => {
      it('should match snapshot', () => {
        const BurgerShapeTree = renderer.create(<BurgerShape/>).toJSON()
        expect(BurgerShapeTree).toMatchSnapshot()
      })
    })
    describe('ButtonText', () => {
      it('should match snapshot', () => {
        const ButtonTextTree = renderer.create(<ButtonText/>).toJSON()
        expect(ButtonTextTree).toMatchSnapshot()
      })
    })
  })
  describe('components rendering', () => {
    const children = 'Hello'
    it('should render BurgerShape and ButtonText with children', () => {
      const ButtonBurgerShapeWrapper = shallow(<ButtonBurgerShape>{children}</ButtonBurgerShape>)
      expect(ButtonBurgerShapeWrapper.find(BurgerShape)).toHaveLength(1)
      expect(ButtonBurgerShapeWrapper.find(ButtonText)).toHaveLength(1)
      expect(ButtonBurgerShapeWrapper.find(ButtonText).text()).toBe(children)
    })
    it('should call preventDefault if there is no props onClick passed', () => {
      const event = {
        preventDefault: jest.fn()
      }
      const ButtonBurgerShapeWrapper = shallow(<ButtonBurgerShape>{children}</ButtonBurgerShape>)
      ButtonBurgerShapeWrapper.find(BurgerShape).simulate('click', event)
      expect(event.preventDefault).toHaveBeenCalledTimes(1)
    })
    it('should call props onClick if there is one', () => {
      const props = {
        onClick: jest.fn()
      }
      const ButtonBurgerShapeWrapper = shallow(<ButtonBurgerShape {...props}>{children}</ButtonBurgerShape>)
      ButtonBurgerShapeWrapper.find(BurgerShape).simulate('click', {})
      expect(props.onClick).toHaveBeenCalledTimes(1)
    })
  })
})
