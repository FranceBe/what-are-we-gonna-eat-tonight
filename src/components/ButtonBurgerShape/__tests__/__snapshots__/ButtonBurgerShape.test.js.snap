// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBurgerShape styled components BurgerShape should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 50px;
  width: 50px;
  border-radius: 50px 50px 10px 10px;
  background: linear-gradient(0deg,#EBA789 15%,#50C92C 27%,#EE6663 47%,#F6FA19 45%,#50C92C 60%,#EBA789 79%);
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
/>
`;

exports[`ButtonBurgerShape styled components ButtonText should match snapshot 1`] = `
.c0 {
  background-color: rgba(255,255,255,0.7);
  padding: 5px;
  margin-bottom: 5px;
  font-size: 0.875rem;
}

<span
  className="c0"
/>
`;
